test/hardhat/utils/bitcoin-utils.js

```
@@ -0,0 +1,290 @@
+   * @param {number} confirmations - The number of confirmations to wait for
+   * @returns {Promise<number>} - The number of confirmations
+   */
+  async waitForConfirmation(txid, confirmations = 1) {
+    console.log(`Waiting for ${confirmations} confirmation(s) for tx: ${txid}`);
+    
+    while (true) {
+      try {
+        const response = await axios.get(`${this.esploraApiUrl}/api/tx/${txid}`);
+        const tx = response.data;
+        
+        if (tx.status && tx.status.confirmed) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)